Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [ ]:
dic={'time_in_hr':[5,6,4,8,7,10,9],
'exam_score':[45,56,34,78,70,95,88]}

In [ ]:
import pandas as pd

In [ ]:
df=pd.DataFrame(dic)

In [ ]:
df

time_in_hr  exam_score
0           5          45
1           6          56
2           4          34
3           8          78
4           7          70
5          10          95
6           9          88

In [ ]:
import seaborn as sns

In [ ]:
df1.corr(method='spearman')

Sleeptime_in_hr  Job_satisfaction
Sleeptime_in_hr          1.000000          0.143795
Job_satisfaction         0.143795          1.000000

In [ ]:
df.corr()

time_in_hr  exam_score
time_in_hr    1.000000    0.996155
exam_score    0.996155    1.000000

Since the correlation is positive and the value is towards 1 so th both data is highly correleted

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.

In [ ]:
dic1={'Sleeptime_in_hr':[5,6,5,8,7,8,9,5,9,8],
'Job_satisfaction':[2,5,9,4,9,3,2,8,1,3]}

In [ ]:
df1=pd.DataFrame(dic1)

In [ ]:
df1

Sleeptime_in_hr  Job_satisfaction
0                5                 2
1                6                 5
2                5                 9
3                8                 4
4                7                 9
5                8                 3
6                9                 2
7                5                 8
8                9                 1
9                8                 3

As we can see clearly that data is +ve but is in more towards -1  and less towards +1 so we can say both data is not some how correlated, for some employee sleep hr may be the factor for job satisfaction but not for all employee some may have different factor as well

Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient coefficient and the Spearman's rank correlation
between these two variables and compare the results.


In [75]:
import numpy as np
import pandas as pd
hours_of_exercise=np.random.normal(5,2,50)
BMI=np.random.normal(30,4,50)
df3=pd.DataFrame({
    'hours_of_exercise':hours_of_exercise,
    'BMI':BMI})


In [76]:
df3.head()

hours_of_exercise        BMI
0           3.432209  28.462056
1           4.846091  33.826326
2           5.577891  33.619155
3           6.771149  23.776805
4           1.312057  24.416841

In [77]:
df3.corr('pearson')

hours_of_exercise       BMI
hours_of_exercise           1.000000  0.195886
BMI                         0.195886  1.000000

In [79]:
df3.corr(method='spearman')

hours_of_exercise       BMI
hours_of_exercise           1.000000  0.158223
BMI                         0.158223  1.000000

The Pearson correlation coefficient measures the linear relationship between two variables, and it ranges from -1 (perfect negative correlation) to +1 (perfect positive correlation), with 0 indicating no correlation. In this case, the Pearson correlation coefficient between num_of_hours and bmi is 0.195886, which indicates a weak positive correlation between the two variables.

On the other hand, the Spearman correlation coefficient measures the monotonic relationship between two variables, which means it assesses the strength and direction of a relationship between two variables, without assuming any particular functional form of the relationship. It also ranges from -1 to +1, with 0 indicating no correlation. In this case, the Spearman correlation coefficient between num_of_hours and bmi is 0.158223, which also indicates a weak positive correlation between the two variables.

Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.

In [84]:
num_of_hours = np.random.normal(8, 3, 50)
amount_of_activity= np.random.normal(12, 5, 50)
df = pd.DataFrame({
    'num_of_hours': num_of_hours,
    'physical_activity': amount_of_activity
})
df.head()

num_of_hours  physical_activity
0     10.426008          10.095039
1     10.980778          11.217046
2     12.185925          19.951227
3      9.160047          22.274384
4     12.149576          26.489939

In [85]:
df.corr('pearson')

num_of_hours  physical_activity
num_of_hours           1.000000           0.210297
physical_activity      0.210297           1.000000

Q5. A survey was conducted to examine the relationship between age and preference for a particular
brand of soft drink. The survey results are shown below:

In [23]:
data={
    'Age':[25,42,37,19,31,28],
    'Soft_drink_Preference':['Coke','Pepsi','Mountain dew','Coke','Pepsi','Coke']
}

In [43]:
import pandas as pd

In [44]:
df2=pd.DataFrame(data)

In [56]:
from sklearn.preprocessing import LabelEncoder

In [57]:
encoding=LabelEncoder()

In [60]:
import numpy as np
encoded=encoding.fit_transform(df2[['Soft_drink_Preference']])

/usr/local/lib/python3.10/dist-packages/sklearn/preprocessing/_label.py:116: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [62]:
df=pd.DataFrame(encoded,columns=['Soft_drink_Preference'])

In [65]:
newdata=pd.concat([df2,df],axis=1)

In [66]:
newdata

Age  Soft_drink_Preference  Soft_drink_Preference
0   25                   Coke                      0
1   42                  Pepsi                      2
2   37           Mountain dew                      1
3   19                   Coke                      0
4   31                  Pepsi                      2
5   28                   Coke                      0

In [68]:
newdata.corr('pearson')

<ipython-input-68-2616d61f9d8d>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  newdata.corr('pearson')


Age  Soft_drink_Preference
Age                    1.000000               0.769175
Soft_drink_Preference  0.769175               1.000000

Hence we can see age has 0.76 correlation which is near 1 so it is higly correlated

Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [86]:
call_per_day=np.random.normal(15,3,30)
call_per_week=np.random.normal(65,3,30)

In [87]:
df4=pd.DataFrame({
    'call_per_day':call_per_day,
    'call_per_week':call_per_week
})

In [88]:
df4

call_per_day  call_per_week
0      20.664704      65.464625
1      17.515626      66.926271
2      18.865558      67.430396
3      13.172175      63.471921
4      18.534094      66.285090
5      15.769867      60.096767
6       8.517857      63.775825
7      16.100468      67.829162
8      16.289902      70.287491
9      11.086195      59.727100
10     15.931143      59.100255
11     15.974039      65.019911
12     10.598183      62.911673
13     14.114747      67.734086
14      7.685435      74.609326
15     15.067910      61.985704
16     11.211112      67.769879
17     16.185743      62.979617
18     13.477818      65.500461
19     16.278712      64.320118
20     15.124744      68.682472
21     10.615243      66.437649
22     16.741889      69.345623
23     14.275634      71.124167
24     10.145359      66.300664
25     16.428741      61.772891
26     14.176049      67.046840
27     16.577822      64.994750
28     16.637241      65.075587
29      9.995207      69.283039

In [89]:
df4.corr('pearson')

call_per_day  call_per_week
call_per_day       1.000000      -0.151245
call_per_week     -0.151245       1.000000

In [90]:
df4.corr('spearman')

call_per_day  call_per_week
call_per_day       1.000000      -0.033593
call_per_week     -0.033593       1.000000